Mathlib/MeasureTheory/Integral/Lebesgue.lean

```diff
@@ -1016,6 +1028,71 @@ theorem set_lintegral_strict_mono {f g : α → ℝ≥0∞} {s : Set α} (hsm :
   lintegral_strict_mono (by simp [hs]) hg.aemeasurable hfi ((ae_restrict_iff' hsm).mpr h)
 #align measure_theory.set_lintegral_strict_mono MeasureTheory.set_lintegral_strict_mono
 
+/-- If the integral of two measurable and finite functions `f` and `g` are finite on
+every subsets, then
+`f =ᵐ[μ] g ↔ ∀ ⦃s⦄, MeasurableSet s → ∫⁻ x in s, f x ∂μ = ∫⁻ x in s, g x ∂μ` -/
+theorem set_lintegral_eq_iff_ae_eq {f g : α → ℝ≥0∞} (hf : Measurable f) (hg : Measurable g)
```

Comment:
We already have many versions of similar theorems in the file `MeasureTheory.Function.AEEqOfIntegralEq`, see for instance `AEMeasurable.ae_eq_of_forall_set_lintegral_eq`. How does your statement compare to these?